plaso/parsers/aws_elb_access.py

```diff
@@ -0,0 +1,323 @@
+      _INTEGER.setResultsName('sent_bytes') +
+      pyparsing.quotedString.setResultsName('request')
+          .setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName('user_agent')
+          .setParseAction(pyparsing.removeQuotes) +
+      _WORD.setResultsName('ssl_cipher') +
+      _WORD.setResultsName('ssl_protocol') +
+      _WORD.setResultsName('target_group_arn') +
+      _WORD.setResultsName('trace_identifier') +
+      pyparsing.quotedString.setResultsName(
+          'domain_name').setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName(
+          'chosen_cert_arn').setParseAction(pyparsing.removeQuotes) +
+      _INTEGER.setResultsName('matched_rule_priority') +
+      _DATE_TIME_ISOFORMAT_STRING.setResultsName('request_creation_time') +
+      pyparsing.quotedString.setResultsName(
+          'actions_executed').setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName(
+          'redirect_url').setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName(
+          'error_reason').setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName(
+          'target_port_list').setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName(
+          'target_status_code_list').setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName(
+          'classification').setParseAction(pyparsing.removeQuotes) +
+      pyparsing.quotedString.setResultsName(
+          'classification_reason').setParseAction(pyparsing.removeQuotes)
+  )
+
+  LINE_STRUCTURES = [('elb_accesslog', _LOG_LINE)]
+
+  def _GetValue(self, structure, name, key_name):
+    """Retrieves a value from a Pyparsing.Group structure
+
+    Args:
+      structure (pyparsing.ParseResults): tokens from a parsed log line.
+      name (str): name of the token.
+      key_name (str): key name to retrieve the value of.
+
+    Returns:
+      object: value for the specified key.
+      """
+    structure_value = self._GetValueFromStructure(structure, name)
+    return structure_value.get(key_name)
+
+  def ParseRecord(self, parser_mediator, key, structure):
+    """Parses a log record structure and produces events.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfVFS.
+      key (str): name of the parsed structure.
+      structure (pyparsing.ParseResults): structure parsed from the log file.
+
+    Raises:
+      ParseError: when the structure type is unsupported.
+    """
+
+    if key != 'elb_accesslog':
+      raise errors.ParseError(
+          'Unable to parse record, unknown structure: {0:s}'.format(key))
+
+    time_elements_structure = structure.get('time')
+    if time_elements_structure:
+      year, month, day_of_month, hours, minutes, seconds, microseconds = (
```

Comment:
looks like time_elements_structure is in the right format to pass to `dfdatetime_time_elements.TimeElements` why convert it into time_elements_tuple?

Reply:
I changed the way this is handled to:

```
    try:
      date_time = dfdatetime_time_elements.TimeElements()
      date_time.CopyFromStringISO8601(time_elements_structure)
    except ValueError:
      parser_mediator.ProduceExtractionWarning(
          'invalid date time value: {0!s}'.format(time_elements_structure))
      return
```